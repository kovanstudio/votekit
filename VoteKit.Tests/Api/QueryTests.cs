using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using HotChocolate;
using HotChocolate.Execution;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.Extensions.DependencyInjection.Extensions;
using Snapshooter.Xunit;
using VoteKit.Api;
using Xunit;

namespace VoteKit.Tests.Api;

public class QueryTests : IAsyncLifetime
{
  private IRequestExecutor _requestExecutor;

  [Fact]
  public async Task Query_Config()
  {
    var res = await _requestExecutor.ExecuteAsync("{ config { basePath, emptyGuid } }");
    res.MatchSnapshot();
  }

  public async Task InitializeAsync()
  {
    var config = new ConfigurationBuilder().AddInMemoryCollection(new Dictionary<string, string>
    {
      { "HTTP:BasePath", "http://localhost:3000/" }
    }).Build();

    _requestExecutor = await new ServiceCollection()
      .AddSingleton<IConfiguration>(config)
      .AddGraphQL()
      .BuildApi()
      .BuildRequestExecutorAsync();
  }

  public Task DisposeAsync()
  {
    return Task.CompletedTask;
  }
}
