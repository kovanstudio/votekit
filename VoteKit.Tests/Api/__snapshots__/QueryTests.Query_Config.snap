{
  "data": {
    "config": {
      "basePath": "http://localhost:3000/",
      "emptyGuid": "00000000-0000-0000-0000-000000000000"
    }
  }
}
